ledger-core/v2/virtual/handlers/vdelegatedcallrequest_test.go

```diff
@@ -0,0 +1,427 @@
+
+		caller      = gen.UniqueReference()
+		outgoing    = gen.UniqueID()
+		requestRef  = reference.NewSelf(gen.UniqueIDWithPulse(pulse.OfNow()))
+		objectRef   = reference.NewSelf(outgoing)
+		bargeIn     = smachine.BargeIn{}
+		sharedState = &object.SharedState{
+			Info: object.Info{
+				PendingTable:                     object.NewRequestTable(),
+				OrderedPendingEarliestPulse:      pulse.OfNow() - 100,
+				ActiveOrderedPendingCount:        1,
+				KnownRequests:                    object.NewRequestTable(),
+				ReadyToWork:                      smsync.NewConditional(1, "ReadyToWork").SyncLink(),
+				OrderedExecute:                   smsync.NewConditional(1, "MutableExecution").SyncLink(),
+				OrderedPendingListFilledCallback: bargeIn,
+			},
+		}
+		callFlags = payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty)
+	)
+
+	slotMachine := slotdebugger.New(ctx, t, false)
+	slotMachine.PrepareMockedMessageSender(mc)
+
+	expectedResponse := &payload.VDelegatedCallResponse{
+		DelegationSpec: payload.CallDelegationToken{
+			TokenTypeAndFlags: payload.DelegationTokenTypeCall,
+			Approver:          nodeRef,
+			DelegateTo:        caller,
+			PulseNumber:       pulse.OfNow(),
+			Callee:            objectRef,
+			Caller:            caller,
+			ApproverSignature: []byte{0xde, 0xad, 0xbe, 0xef},
+		},
+	}
+
+	var jetMock jet.AffinityHelper = jet.NewAffinityHelperMock(mc).MeMock.Return(nodeRef)
+
+	slotMachine.AddInterfaceDependency(&jetMock)
+
+	smDelegatedCallRequest := SMVDelegatedCallRequest{
+		Payload: &payload.VDelegatedCallRequest{
+			RequestReference: requestRef,
+			CallFlags:        callFlags,
+			Callee:           objectRef,
+		},
+		Meta: &payload.Meta{
+			Sender: caller,
+		},
+	}
+	catalogWrapper := object.NewCatalogMockWrapper(mc)
+
+	{
+		var catalog object.Catalog = catalogWrapper.Mock()
+		slotMachine.AddInterfaceDependency(&catalog)
+
+		sharedStateData := smachine.NewUnboundSharedData(sharedState)
+		smObjectAccessor := object.SharedStateAccessor{SharedDataLink: sharedStateData}
+
+		catalogWrapper.AddObject(objectRef, smObjectAccessor)
+		catalogWrapper.AllowAccessMode(object.CatalogMockAccessGetOrCreate)
+	}
+
+	slotMachine.Start()
+	defer slotMachine.Stop()
+
+	slotMachine.MessageSender.SendTarget.Set(func(_ context.Context, msg payload.Marshaler, target reference.Global, _ ...messagesender.SendOption) error {
+		res := msg.(*payload.VDelegatedCallResponse)
+		// ensure that both times request is the same
+		assert.Equal(t, caller, target)
+		assert.Equal(t, expectedResponse, res)
+		return nil
+	})
+
+	smWrapper := slotMachine.AddStateMachine(ctx, &smDelegatedCallRequest)
+
+	slotMachine.RunTil(smWrapper.AfterStep(smDelegatedCallRequest.stepBuildResponse))
+
+	assert.True(t, sharedState.PendingTable.GetList(contract.CallTolerable).Exist(requestRef))
+
+	require.NoError(t, catalogWrapper.CheckDone())
+	mc.Finish()
+}
+
+func TestSMVDelegatedCallRequest_TooOld(t *testing.T) {
+	t.Log("C4984")
+	var (
```

Comment:
is it possible please, to reuse a local function for initialization of tests?

Reply:
refactored tests to table tests